conman/redirects/models.py

```diff
@@ -13,3 +15,9 @@ class RouteRedirect(Route):
     handler = handlers.RouteRedirectHandler.path()
     target = models.ForeignKey('routes.Route', related_name='+')
     permanent = models.BooleanField(default=False, blank=True)
+
+    def save(self, *args, **kwargs):
+        """Forbid setting target equal to self."""
+        if self.target_id == self.route_ptr_id:
+            raise ValidationError(_('A RouteRedirect cannot redirect to itself.'))
+        return super().save(*args, **kwargs)
```

Comment:
Yeah, it looks like [that's the place to do it](https://docs.djangoproject.com/en/1.8/topics/forms/modelforms/#interaction-with-model-validation)